src/main/scala/is/hail/expr/ir/IR.scala

```diff
@@ -88,9 +88,7 @@ final case class ToSet(a: IR) extends InferIR
 final case class ToDict(a: IR) extends InferIR
 final case class ToArray(a: IR) extends InferIR
 
-final case class SetContains(set: IR, elem: IR) extends IR { val typ: Type = TBoolean() }
-final case class DictContains(set: IR, elem: IR) extends IR { val typ: Type = TBoolean() }
-final case class DictGet(dict: IR, key: IR) extends InferIR
+final case class SearchOrderedCollection(orderedCollection: IR, elem: IR, onKey: Boolean) extends IR { val typ: Type = TInt32() }
```

Comment:
It might be helpful if the name specified if this returns the largest index <= the key, or the smallest index >= the key. I know I would have to go look every time. The C++ standard library calls these `lower_bound` and `upper_bound`, which I think is pretty clear. But I could be dissuaded.